TRANSITION.md

```diff
@@ -0,0 +1,363 @@
+from girder.plugins.worker import utils
+
+# Define the analysis
+girder_worker_run_analysis = {
+    'name': 'add',
+    'inputs': [
+        {'name': 'a', 'type': 'integer', 'format': 'integer', 'default':
+         {'format': 'json', 'data': '0'}},
+        {'name': 'b', 'type': 'integer', 'format': 'integer'}
+    ],
+    'outputs': [{'name': 'c', 'type': 'integer', 'format': 'integer'}],
+    'script': 'c = a + b',
+    'mode': 'python'}
+
+# Define the inputs
+girder_worker_run_inputs = {'a': {'format': 'integer', 'data': 1},
+                            'b': {'format': 'integer', 'data': 2}}
+
+# Define the outputs
+girder_worker_run_outputs = {'c': {'format': 'integer'}}
+
+# Create the job
+job = Job().createJob(
+    title='Add a and b',
+    handler='worker_handler',
+    user=self.getCurrentUser(), 
+	public=False, 
+	args=(girder_worker_run_analysis,),
+    kwargs={'inputs': self.girder_worker_run_inputs,
+            'outputs': self.girder_worker_run_outputs})
+
+# Set up the jobInfoSpec
+job['kwargs']['jobInfo'] = utils.jobInfoSpec(job)
+
+# Save the Job
+Job().save(job)
+
+# Schedule the Job
+Job().scheduleJob(job)
+
+return job
+
+```
+
+All of this is necessary to execute the script ```c = a + b``` on the arguments ```a = 1``` and ```b = 2```.  Assuming that we have a task function ```my_task``` in ```my_package.tasks```  with following code: 
+
+```python
+from girder_worker.app import app
+from girder_worker.utils import girder_job
+
+@girder_job(title='Add a and b')
+@app.task(bind=True)
+def my_task(self):
+	return a + b
+```
+
+Then the legacy code above may be rewritten as follows:
+
+```python
+from my_package.tasks import example_task
+
+async_result = my_task.delay(1, 2)
+
+return async_result.job
+```
+
+All ```Job``` related code is removed in favor of importing the task directly,  and using the [Celery API for calling tasks](https://github.com/girder/girder_worker/blob/transition-document/HACKING.md#the-celery-method).  Girder Worker has been refactored to move all job related code inside the application. Various knobs and dials have been exposed to allow modifying aspects of the Girder Job that is created. For instance you may set the title of the job at call time by using the ```girder_job_title``` special keyword argument:
+
+```python
+from my_package.tasks import example_task
+
+async_result = my_task.delay(
+    1, 2, girder_job_title="A Custom Job Title")
+
+return async_result.job
+```
+
+For more details about manipulating Job attributes,  please read [The Celery Method](https://github.com/girder/girder_worker/blob/transition-document/HACKING.md#the-celery-method) section of the HACKING document.
+
+
+### Transformations
+
+The following steps are sufficient for converting legacy Girder Worker tasks into modern Celery based tasks. If however your task requires access to information inside Girder,  then it is recommended that you include "Transform" classes with your package.
```

Comment:
I wonder if most of this section should be made the official "Transforms" documentation in another location, and that this document should just point to it. When porting, I expect the "80%" case of transforms will already be available in the utils package, so all people don't need to know the internal mechanics of how they are implemented.